### DISCUSSION TITLE
Port forwarding broken

### DISCUSSION THREAD

**Discussion by NottPeak** [community/community#69926] [Codespaces] [answered]
### Select Topic Area

Question

### Body

At school, I use codespaces for some projects, most of which require port forwarding. I am using node as a backend and just simple html/css/js frontend, nothing special. Whenever I run my service, github doesn't pick up the port in the terminal and doesn't port forward automatically, and whenever I try to manually port forward, I receive this error: `Unable to forward localhost:8080. Network Error`. I am wondering if this could be due to some network block at the school, as I can port forward normally on my home desktop, or maybe it could be a filter installed on the school device. Either way, would anyone know a way to diagnose this without developer tools? It would be much easier if I could for example see the network requests or console to figure this out, but they are disabled as it is a school managed device and they don't want any shenanigans. Right now my main assumption is a request being sent from my laptop is being blocked by a filter or something and doesn't get sent to the servers, resulting in the network error. I can contact the IT admins if I am able to get a IP or something similar that might need to get whitelisted on a firewall. Sorry for the lengthy paragraph, I am not great at explaining things in short sentences

Tldr; On my school device (and network, not sure which one it is) I am unable to port forward, what would my IT admins need to whitelist in order to allow my device to send the port forwarding request to the github servers? 

PS: All other parts of github codespaces works, terminal works, extensions work, everything I have tried besides port forwarding has worked.

**Comment by thevindu-w** [comment, marked as answer]
Probably you are behind a NAT or a firewall. If that is the case, you have to contact the IT admins of your school.
If you are trying to expose a web service, there are alternatives like [ngrok](https://ngrok.com/) and [localhost.run](https://localhost.run/) which does not require you to allow the port through the firewall. There are many services like this.

**Reply by NottPeak** [reply]
I do think it is a firewall issue, but due to restrictions I am not sure what endpoint needs to be whitelisted for port forwarding to work, and my IT is very busy so I don't think they would have time to do some troubleshooting.

**Reply by thevindu-w** [reply]
Yes, but the [ngrok](https://ngrok.com/), [localhost.run](https://localhost.run/) and other alternatives doesn't require the IT admins to whitelist ports. So, you can try them first, if you don't like to directly contact IT admins.

**Reply by NottPeak** [reply]
I would be completely fine contacting the IT admins if I have something to give them, but those are interesting alternatives, but do they have node.js support? My project requires node, so I would only be able to work on design without it

**Reply by thevindu-w** [reply]
It does not matter whether it's node or something else. Also they don't require you to change your code.
For an example, if you choose ngrok, you can follow the steps in [https://ngrok.com/docs/getting-started/#step-2-install-the-ngrok-agent](https://ngrok.com/docs/getting-started/#step-2-install-the-ngrok-agent) starting from step 2 (step 1 is to create an HTTP server which you already have in node).
[localhost.run](https://localhost.run/docs/) is even simpler. It just require ssh. Refer [https://localhost.run/docs/](https://localhost.run/docs/)

**Reply by NottPeak** [reply]
Ok, I will check those out. If anyone could find what needs to be whitelisted for anyone else with this issue (I saw a bunch of people complaining) that would be great, but I think this solves my personal issue

**Reply by thevindu-w** [reply]
If that solved the issue, please mark the answer as accepted so that future readers can find it easily.
